### PR TITLE
forwarder: return NXDomain on e.is_nx_domain()

### PR DESCRIPTION
**Pull request by balboah**
`ZoneType::Forward` authority seems to not have gotten much love. This adds NXDomain status on "not found" errors.

We're still missing a SOA record, perhaps this can be picked up on a later refactor to re-use parts of `AuthorityLookupState`.

### CODE REVIEW

**Comment by bluejekyll** [comment]
Do you think we should add a tear for this? The integration tests in this area are a little light, but it should be possible?

**Comment by balboah** [comment]
Tests are always good, but I'm not sure what the correct behaviour is :) I believe forwarder mode is mostly correct when it gets records to show, but on errors I know that the SOA record should appear and I saw that this seems to be handled when in Master/Slave mode.
When comparing with other servers I noticed that authority status should also be set when records are missing, currently it's always 0.

Not exactly sure how high priority it is, I haven't experienced any issues because of it missing

**Comment by bluejekyll** [comment]
I see. To get the "correct" behavior we'd need to go back to the RFC. I think SOA is only required if it's an authoritative request. It might be worth reviewing this: https://tools.ietf.org/html/rfc1034#section-4.3.4

I *think* we don't need to add SOA on forward, but we can. I think one of the negative things here, is that what we really want to do is return the actual upstream NXDOMAIN response, and currently the Resolver is not returning that in this error. It would mean going back into the resolver and making sure we return that with the error I think.

Basically, I think this is a good change, we should merge it. I also wonder what we're doing in the NOERROR but no records returned case as well.

Anyway, as testing goes, there is this: https://github.com/bluejekyll/trust-dns/blob/master/crates/server/tests/forwarder.rs#L19

I'm not sure why it's ignored right now, might be racy due to startup or something.

**Comment by balboah** [comment]
I'm currently not using the resolver, but returning Lookup/LookupError created by other means.
I agree it would be nice if we could return whatever Lookup/Error we like that the server should return to the client in the appropriate format according to RFC's.

**Comment by balboah** [comment]
rebased on latest master, feel free to merge
